lib/nodes/addon/components/driver-azure/component.js

```diff
@@ -165,6 +303,151 @@ export default Component.extend(NodeDriver, {
     return false;
   }),
 
+  selectedVmSizeHasZones: computed('config.{location,size}', 'value.size', 'vmSizes', 'vmsWithAvailabilityZones', function() {
+    const dataForSelectedSize = this.vmsWithAvailabilityZones.filter((vmData) => {
+      const { Name } = vmData;
+
+      return Name === get(this, 'config.size');
+    });
+
+    if (dataForSelectedSize.length > 0) {
+      return dataForSelectedSize[0].AvailabilityZones.length > 0;
+    }
+
+    return false;
+  }),
+
+  selectedVmSizeSupportsAN: computed('config.{location,size}', 'vmSizes', 'vmsWithAcceleratedNetworking', function() {
+    const selectedSizeIsValid = !!this.vmsWithAcceleratedNetworking.find((vmData) => {
+      return get(this, 'config.size') === vmData.Name;
+    });
+
+    return selectedSizeIsValid;
+  }),
+
+  vmSizeAcceleratedNetworkingWarning: computed('config.{acceleratedNetworking,location,size}', 'selectedVmSizeSupportsAN', 'vmSizes.length', function() {
+    if (get(this, 'vmSizes.length') === 0) {
+      // Don't show the warning until the VM sizes are loaded
+      return '';
+    }
+
+    const intl = window.l('service:intl')
+
+    if (!this.selectedVmSizeSupportsAN && get(this, 'config.acceleratedNetworking')) {
+      return intl.t('nodeDriver.azure.size.selectedSizeAcceleratedNetworkingWarning');
+    }
+
+    return '';
+  }),
+
+  vmSizeAvailabilityWarning: computed('config.{location,size}', 'regions.length', 'selectedVmSizeExistsInSelectedRegion', 'vmSizes.length', function() {
+    if (get(this, 'regions.length') === 0) {
+      // Don't show the warning until the regions are loaded
+      return '';
+    }
+    if (get(this, 'vmSizes.length') === 0) {
+      // Don't show the warning until the VM sizes are loaded
+      return '';
+    }
+    const selectedVmIsAvailableInSelectedRegion = this.vmSizes.filter((vmData) => {
+      return vmData.Name === get(this, 'config.size');
```

Comment:
This looks like the same thing as `selectedVmSizeExistsInSelectedRegion`

Reply:
I agree - replaced with:

```js
if (!get(this, 'selectedVmSizeExistsInSelectedRegion')) {
      return intl.t('nodeDriver.azure.size.availabilityWarning');
    }
```